### DISCUSSION TITLE
Welcome to Giotto Discussions!

### DISCUSSION THREAD

**Discussion by RubD** [giotto-suite/Giotto#86] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by hhrncir3** [comment]
Hi all, 

This is likely a small issue, but I cannot figure it out no matter how many things I change. When I try to upload the practice dataset Visium Kidney Data, I get the following error: 

**Code:** visium_kidney = createGiottoVisiumObject(visium_dir = data_path, expr_data = 'raw',
                                                                                  png_name = 'V1_Mouse_Kidney_image.png',
                                                                                    gene_column_index = 2, instructions = instrs),
 **Error:** A structured visium directory will be used 
 Error in createGiottoVisiumObject(visium_dir = data_path, expr_data = "raw",  : 
                                     /Users/hhrnc/OneDrive/Desktop/Gracz Lab/Rebus/V1_Mouse_Kidney_fastqs/V1_Mouse_Kidney_fastqs/ does not exist 

Any guesses on what is going wrong?

**Comment by RubD** [comment]
Hi,

This looks like the path to the visium directory does not exist. 

You can download the data from the 10X genomics website and then the directory itself should look like this:

![Screen Shot 2021-09-29 at 15 06 29](https://user-images.githubusercontent.com/8243389/135332965-00b862bf-d12d-48f3-a679-3d3c9e48584d.png)

Your path is probably not the right one: **/Users/hhrnc/OneDrive/Desktop/Gracz Lab/Rebus/V1_Mouse_Kidney_fastqs/V1_Mouse_Kidney_fastqs/**

If this is your own dataset and you want to use the raw fastq files, then you first need to process them with the spaceranger tool from 10x. The final output will be a structured directory that you can read in directly within Giotto.

**Reply by hhrncir3** [reply]
That helps a lot. I think I am having issues setting up the Visium 10X Directory for the sample data. Is there a way to download the entire directory? I have been downloading the directory one piece at a time... What does the path to your directory look like? Thanks Ruben!

**Reply by RubD** [reply]
For most dataset examples we have the function **getSpatialDataset**, which automatically downloads all the files that you need to run that dataset example. However, for the visium examples from 10X we didn't do that, because they are all easily available from the 10X website. And they also provide instructions on how to download the total dataset. If you would download that dataset to your Desktop directory, then you path would probably look something like this: /Users/rubendries/Desktop/Kidney_data. This is assuming you're working on a Mac and the name of the 10X visium folder is called Kideny_data.

**Reply by hhrncir3** [reply]
Hi Ruben, 

I was unable to get the Visium data to read the directory correctly, but since it was just practice, I switched to the Codex spleen practice dataset. This data set works great for me. I am now analyzing my own data. Is there a way to change the axis (I want to only look at one section of the graph bc my coordinates range up to 200,000. I would rather just look at (0, 20000) on my x and y axis. How do I change axis in _spatplot_?

**Reply by fay-onyeogaziri** [reply]
Hi, I encountered the same issue. 
I attched what the 10x site looks like now, could you point me towards how to download the files that they look like the directory you had above instead of individual files?

Thanks,
Favour
<img width="1728" alt="Screenshot 2024-02-29 at 11 48 23" src="https://github.com/drieslab/Giotto/assets/103258692/ceb52025-6147-4163-bb8a-1592acce170b">

**Reply by josschavezf** [reply]
@fay-onyeogaziri  In the window you are showing, click on "Show batch download instructions" then select wget 
Copy the corresponding lines that would look like:

wget https://cf.10xgenomics.com/samples/spatial-exp/1.1.0/V1_Adult_Mouse_Brain/V1_Adult_Mouse_Brain_spatial.tar.gz
wget https://cf.10xgenomics.com/samples/spatial-exp/1.1.0/V1_Adult_Mouse_Brain/V1_Adult_Mouse_Brain_raw_feature_bc_matrix.tar.gz

You have to run these commands in the bash terminal, either from your computer or through the Terminal window in RStudio.

After you download the files, you have to uncompress the folders

**Reply by fay-onyeogaziri** [reply]
Thanks a lot!

**Comment by RubD** [comment]
There are currently two ways you could do that:  
1. return the ggplot object and manually add another layer to set the x-axis. Something like:  

```
plot = spatPlot(..., return_plot = TRUE)
plot = plot + ggplot2::xlim(c(0, 200000))
plot
```

2. subset the Giotto object with the `subsetGiottoLocs()` function and then use spatPlot on the subsetted Giotto Object

I could perhaps add this directly to the spatPlot function in the next version of Giotto.

**Comment by hhrncir3** [comment]
I am excited that I have Giotto working robustly for me now. I am interested at looking at the data for cells that express my marker gene, so I need to subset my data. **I want to remove cells which do not express gene X (my marker gene).** How do I do that?

**Comment by RubD** [comment]
You should be able to do this using a combination of functions:
1. [addGenesPerc](https://rubd.github.io/Giotto_site/reference/addGenesPerc.html) to get information about the percentage of normalized counts for your marker gene in each cell
2. [pDataDT](https://rubd.github.io/Giotto_site/reference/pDataDT.html) will return the cell metadata (or pheno data.table) and allows you to simply subset the data.table for minimum detection percentage of your marker genes. You can find the associated cell IDs and use them in 3.
3. [subsetGiotto](https://rubd.github.io/Giotto_site/reference/subsetGiotto.html) you can provide the cell IDs obtained in 2 to subset only these cells that show expression of your marker gene


This is all assuming that you're using the current master branch of Giotto.
